File: app/src/main/java/com/example/myapplication/DTO/CarAnnouncementsRequestDTO.java
package com.example.myapplication.DTO;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.firebase.database.PropertyName;

import java.util.List;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import lombok.extern.jackson.Jacksonized;

@Jacksonized
@AllArgsConstructor
@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
@Getter
@ToString
public class CarAnnouncementsRequestDTO {
    @JsonProperty("brand")
    @Getter
    List<String> brand;

    @JsonProperty("model")
    @Getter
    List<String> model;

    @JsonProperty("transmission")
    @Getter
    List<Integer> transmission;

    @JsonProperty("gear")
    @Getter
    List<Integer> gear;

    @JsonProperty("minEngineCapacity")
    @Getter
    Integer minEngineCapacity;

    @JsonProperty("maxEngineCapacity")
    @Getter
    Integer maxEngineCapacity;

    @JsonProperty("minEnginePower")
    @Getter
    Integer minEnginePower;

    @JsonProperty("maxEnginePower")
    @Getter
    Integer maxEnginePower;

    @JsonProperty("color")
    @Getter
    List<Integer> color;

    @JsonProperty("mileage")
    @Getter
    List<String> mileage;

    @JsonProperty("performance")
    @Getter
    List<Integer> performance;
}

File: app/src/main/java/com/example/myapplication/DTO/CarAnnouncementsRequestDTO.kt
package com.example.myapplication.DTO

data class CarAnnouncementsRequestDTO(
    var brand: List<String>? = null,
    var model: List<String>? = null,
    var transmission: List<Int>? = null,
    var gear: List<Int>? = null,
    var minEngineCapacity: Int? = null,
    var maxEngineCapacity: Int? = null,
    var minEnginePower: Int? = null,
    var maxEnginePower: Int? = null,
    var color: List<Int>? = null,
    var mileage: List<String>? = null,
    var performance: List<Int>? = null,
    var description: List<String>? = null,
    val minPrice: Long? = null,
    val maxPrice: Long? = null,
    val state: AnnouncementState? = null
)
